plugins/default/src/Ide/Plugin/ModuleName.hs

```diff
@@ -1,222 +1,206 @@
-    let maybePrefix = listToMaybe . filter (`isPrefixOf` mdlPath) $ paths
-    out state ["prefix", show maybePrefix]
-
-    let maybeMdlName =
-            (\prefix ->
-                intercalate "."
-                . splitDirectories
-                . drop (length prefix + 1)
-                $ dropExtension mdlPath
-            )
-            <$> maybePrefix
-    out state ["mdlName", show maybeMdlName]
-    return $ T.pack <$> maybeMdlName
-  else return $ Just "Main"
+pathModuleName state normFilePath filePath
+    | isLower (head $ takeFileName filePath) = return $ Just "Main"
+    | otherwise = do
+        session :: HscEnvEq <- runAction "ModuleName.ghcSession" state $ use_ GhcSession normFilePath
+        srcPaths <- evalGhcEnv (hscEnvWithImportPaths session) $ importPaths <$> getSessionDynFlags
+        paths <- mapM canonicalizePath srcPaths
+        mdlPath <- canonicalizePath filePath
+        let maybePrefix = find (`isPrefixOf` mdlPath) paths
+
+        let maybeMdlName =
+                ( \prefix ->
+                    intercalate "."
+                        . splitDirectories
+                        . drop (length prefix + 1)
+                        $ dropExtension mdlPath
+                )
+                    <$> maybePrefix
+        return $ T.pack <$> maybeMdlName
 
 -- | The module name, as stated in the module
 codeModuleName :: IdeState -> NormalizedFilePath -> IO (Maybe (Range, Text))
 codeModuleName state nfp =
-  ((\(L (RealSrcSpan l) m) -> (realSrcSpanToRange l, T.pack . show $ m)) <$>)
-    .   join
-    .   (hsmodName . unLoc . pm_parsed_source <$>)
-    <$> runAction "ModuleName.GetParsedModule" state (use GetParsedModule nfp)
+    ((\(L (RealSrcSpan l) m) -> (realSrcSpanToRange l, T.pack . show $ m)) <$>)
+        . ((hsmodName . unLoc . pm_parsed_source) =<<)
+        <$> runAction "ModuleName.GetParsedModule" state (use GetParsedModule nfp)
 
--- | A source code change
-data Action = Action {aUri::Uri,aRange::Range,aTitle::Text,aCode::Text} deriving Show
+-- traceAs :: Show a => String -> a -> a
+-- traceAs lbl a = trace (lbl ++ " = " ++ show a) a
 
--- | Convert an Action to a CodeLens
-asCodeLens :: Text -> Action -> CodeLens
-asCodeLens cid act@Action {..} = CodeLens
-  aRange
-  (Just $ Command aTitle cid (Just (List [toJSON $ asEdit act])))
-  Nothing
-
--- | Convert an Action to a CodeAction
-asCodeAction :: Action -> CAResult
-asCodeAction act@Action {..} = CACodeAction $ CodeAction
-  aTitle
-  (Just CodeActionQuickFix)
-  (Just $ List [])
-  (Just $ asEdit act)
-  Nothing
-
-asEdit :: Action -> WorkspaceEdit
-asEdit act@Action {..} =
-  WorkspaceEdit (Just $ Map.singleton aUri $ List (asTextEdits act)) Nothing
-
-asTextEdits :: Action -> [TextEdit]
-asTextEdits Action {..} = [TextEdit aRange aCode]
-
-out :: IdeState -> [String] -> IO ()
-out state =
-  logPriority (ideLogger state) Debug
-    . pack
-    . unwords
-    . ("Plugin ModuleName " :)
+traceAs :: b -> a -> a
+traceAs _ a = a
```

Comment:
Could you clean these debug functions? Other than that, looks good.

Reply:
Thanks.

But clean them up in what way exactly?

I could redefine them so that they use the standard logger, as in the previous 'out' function.
  
Would that be ok?

Reply:
Yes, that and removing the commented one are what I mean.